docs/syntax-table-destructuring.md

```diff
@@ -0,0 +1,137 @@
+# Table Destructuring Syntax
+
+
+
+## Summary
+Introduce new syntax for unpacking values from tables.
+
+```lua
+local &{ foo, ["@bar"] = bar } = thing
```

Comment:
Not happy about the `&` tbh. If we _really_ had to pick a symbol, `.` would be better.